src/natcap/invest/coastal_vulnerability.py

```diff
@@ -1282,7 +1282,8 @@ def extract_bathymetry_along_ray(
                 'bathymetry input fully cover the fetch ray area?'
                 % (value, {'xoff': ix, 'yoff': iy,
                            'win_xsize': win_size, 'win_ysize': win_size}))
-        if ~numpy.isclose(value, bathy_nodata):
+
+        if bathy_nodata is None or ~numpy.isclose(value, bathy_nodata):

```

Comment:
cool! Is there some resource you have for finding faster alternatives like this?

Reply:
update: `value` is actually an array, not a single value, so `math.isclose` doesn't work :( I changed the variable name to `values` to be less confusing.

Reply:
Ha, sorry to disappoint!  I unfortunately don't know of such a resource.  In this case, I just saw that we were using numpy but then only checking an array with a single value in it.  Numpy's great for operating on arrays that might be very large, but for a single value, it's often faster to just directly compare the values.  Plus, numpy does a [whole lot of extra work](https://github.com/numpy/numpy/blob/e2428594b77aef952610f930bacf4085ef1fcdf9/numpy/core/numeric.py#L2201) to ensure that it can actually compare the values, compared with the python implementation of `isclose`, which is [about as short as you'd expect](https://github.com/python/cpython/blob/82e79480d6e61940d7007d9026fbff0b1a11ad9a/Modules/mathmodule.c#L2932) for this.

Also I was a bit wrong about the timings I had included above.  Since this is a 1-element numpy array, here are the corrected timings:

```python
>>> timeit.timeit('~numpy.isclose(array, 1)', setup='import numpy; array=numpy.array([[1.1]], dtype=numpy.float32)')
42.59300510000003
>>> timeit.timeit('not math.isclose(array[0][0], 1)', setup='import numpy; import math; array=numpy.array([[1.1]], dtype=numpy.float32)')
0.5673721999999941
```

Not _quite_ as fast thanks to the need to do that indexing, but still substantially faster.